### ISSUE TITLE
Is it possible to use js-based config and use gulp?

### ISSUE THREAD

**Issue by melissamcewen** [garris/BackstopJS#236]
Seems like the gulp tasks only utilize json, but it would be useful for them to use js too since that's an option?  Thanks!

**Comment by garris** [comment]
@melissamcewen thanks for posting.   You _can_ use a JS based config... 
https://github.com/garris/BackstopJS#using-a-js-based-config-file-version-100

Is that what you mean?

**Comment by melissamcewen** [comment]
Thanks yes, I set one up, but when I run gulp reference it looks for a .json one

**Comment by garris** [comment]
Please post your console output...

**Comment by melissamcewen** [comment]
```
admins-mbp:backstopjs mmcewen$ gulp reference

Config file not found.

`$ npm run genConfig` generates a handy configuration boilerplate file at: `/Users/mmcewen/Sites/AMA-Corporate-site/docroot/sites/all/themes/ama_theme/backstop.json`. (Will overwrite existing files.)

[14:03:36] Using gulpfile ~/Sites/AMA-Corporate-site/docroot/sites/all/themes/ama_theme/node_modules/backstopjs/gulpfile.js
[14:03:36] Starting 'clean'...
bitmaps_reference was cleaned.
[14:03:36] Starting 'bless'...
[14:03:36] Finished 'bless' after 2.83 ms
[14:03:36] Finished 'clean' after 24 ms
[14:03:36] Starting 'reference'...
gulp.run() has been deprecated. Use task dependencies or gulp.watch task triggering instead.
[14:03:36] Starting 'init'...
[14:03:36] Finished 'init' after 8.08 μs
[14:03:36] Starting 'test'...

Generating reference files.


Running CasperJS with:  [ 'capture/genBitmaps.js' ]
[14:03:36] Finished 'test' after 8.17 ms
reference has run.
[14:03:36] Finished 'reference' after 9.96 ms
CasperJS: CREATING NEW REFERENCE FILES
CasperJS: Clicking button
CasperJS: Clicking button
CasperJS: Clicking button
CasperJS: Ready event received.
CasperJS: Current location is http://getbootstrap.com
CasperJS: onReady.js
CasperJS: remote console > This custom script is running inside your web app!
CasperJS: This custom script is running inside your web app!
remote console > Add your own casper commands here to simulate user interacions or logging in.
Add your own casper commands here to simulate user interacions or logging in.
CasperJS: Screenshots for phone (320x480)
CasperJS: Ready event received.
CasperJS: Current location is http://getbootstrap.com
CasperJS: onReady.js
CasperJS: remote console > This custom script is running inside your web app!
CasperJS: This custom script is running inside your web app!
remote console > Add your own casper commands here to simulate user interacions or logging in.
Add your own casper commands here to simulate user interacions or logging in.
CasperJS: Screenshots for tablet_v (568x1024)
CasperJS: Ready event received.
CasperJS: Current location is http://getbootstrap.com
CasperJS: onReady.js
CasperJS: remote console > This custom script is running inside your web app!
CasperJS: This custom script is running inside your web app!
remote console > Add your own casper commands here to simulate user interacions or logging in.
CasperJS: Add your own casper commands here to simulate user interacions or logging in.
CasperJS: Screenshots for tablet_h (1024x768)
CasperJS: Comparison config file updated.

Bitmap file generation completed.

Run `$ npm run test` to generate diff report.

```

**Comment by melissamcewen** [comment]
I don't see anything looking for a js config file in gulp or a way to pipe in that

**Comment by garris** [comment]
The link I posted above describes how to call a js based config -- you simply add a flag to your command e.g. `npm run reference -- --configPath=~/absolutePathTo/yourTestConfig.js `.     There are multiple examples in the read me.   Is the documentation confusing or did you just miss that?

**Comment by melissamcewen** [comment]
Thanks, when I run on the command line I don't have a problem, it's running through gulp where the flag doesn't seem recognized or available.

```
gulp reference --configPath=/Users/mmcewen/Sites/AMA-Corporate-site/docroot/sites/all/themes/ama_theme/test.js
[14:39:39] Using gulpfile ~/Sites/AMA-Corporate-site/docroot/sites/all/themes/ama_theme/gulpfile.js
[14:39:39] Starting 'reference'...
[14:39:39] Finished 'reference' after 7.55 ms
[gulp-chug] File is a buffer. Need to write buffer to temp file...
[gulp-chug] Writing buffer to node_modules/backstopjs/gulpfile.tmp.1470166779904.js...
[gulp-chug] Spawning process /Users/mmcewen/Sites/AMA-Corporate-site/docroot/sites/all/themes/ama_theme/node_modules/backstopjs/node_modules/gulp/bin/gulp.js with args /Users/mmcewen/Sites/AMA-Corporate-site/docroot/sites/all/themes/ama_theme/node_modules/backstopjs/node_modules/gulp/bin/gulp.js --gulpfile gulpfile.tmp.1470166779904.js reference from directory /Users/mmcewen/Sites/AMA-Corporate-site/docroot/sites/all/themes/ama_theme/node_modules/backstopjs...
[gulp-chug] (node_modules/backstopjs/gulpfile.tmp.1470166779904.js) 
Config file not found.
[gulp-chug] (node_modules/backstopjs/gulpfile.tmp.1470166779904.js) 
`$ npm run genConfig` generates a handy configuration boilerplate file at: `/Users/mmcewen/Sites/AMA-Corporate-site/docroot/sites/all/themes/ama_theme/backstop.json`. (Will overwrite existing files.)
```

**Comment by garris** [comment]
Ah ha.  Now it all makes sense.  Thanks for explaining.  I've never tried to do that -- but this looks like something to try... https://www.npmjs.com/package/gulp-chug#options    Please post back if you can figure it out!  Cheers.

**Comment by melissamcewen** [comment]
I'll try that, thanks!

**Comment by garris** [comment]
Closed by #259
